### PR TITLE
Goals step: Link to the logged-out DIFM flow

### LINKED ISSUE #97461: Task: i1 Loggedout Goals -> DIFM route to existing  difm flow on /start/do-it-for-me
### Details

This iteration is a quick implementation to be able to ship fast and will mean:

LOHP -> Goals and DIFM click -> Redirects to /start/do-it-for-me/new-or-existing-site, which does Account creation -> {DIFM steps, including the DIFM LP} -> Checkout -> Home

See slack discussion. p1734060882124309-slack-C085HCWCEDN. which applies to both Import and DIFM flows.  Same idea of doing a quick implementation.

### Checklist

- [ ]
- [ ]

### Related

_No response_

### PR DESCRIPTION
**Pull request by zaguiini**
Closes https://github.com/Automattic/wp-calypso/issues/97461.

## Proposed Changes

Point the "Let us build a custom site for you" link to "/start/do-it-for-me." The initial idea was to display the "DIFM Intro Screen" before sending the user to the flow, but there are some technical limitations, so we'll redirect them directly for now.

This PR does not attempt to rework the back button or move the Account step after the DIFM Intro Screen. These tasks will be handled at https://github.com/Automattic/wp-calypso/issues/97466.

## Why are these changes being made?

So the `/setup/onboarding` flow becomes more functional.

## Testing Instructions

As a logged-out user, go to `/setup/onboarding`, click the DIFM CTA, verify that you can create an account, click the "Get Started" button in the DIFM Intro Screen, fill out the details, pick a design, and proceed to checkout.

As a logged-in user, go to `/setup/onboarding`, click the DIFM CTA, verify you see the DIFM Intro Screen with the option to pick an existing site or create a new site. Then, you should be able to fill out the details and proceed to checkout.

### CODE REVIEW

File: client/landing/stepper/declarative-flow/onboarding.ts [hunk 1]
```diff
@@ -117,8 +117,7 @@ const onboarding: Flow = {
 							return;
 
 						case SiteIntent.DIFM:
-							// TODO Implement exit to DIFM
-							return;
+							return window.location.assign( '/start/do-it-for-me' );
```
Comment (p-jackson) [inline client/landing/stepper/declarative-flow/onboarding.ts]: We don't have to do it in this PR, but something we need to keep aware of: users might be logged out which means the locale is passed in the URL. Not something we usually have to think about, but we do now that so much is happening before the signup step.

The Spanish onboarding flow starts at `/setup/onboarding/es`. And if that's the case then the link needs to go to `/start/do-it-for-me/es`

**Review by candy02058912 — COMMENTED** [review]
Wondering if directing to https://wordpress.com/website-design-service/ may be acceptable for the time being?

- that’s a logged out landing page (translations are also available)
- minimal effort
- no duplicate screens
- one more link on the goals screen "working"

feels aligned to shipping iteratively, what do y'all think?

**Comment by autumnfjeld** [comment]
This PR will close this issue: https://github.com/Automattic/wp-calypso/issues/97466

**Review by p-jackson — APPROVED** [review]
Everything working well for me in testing.

Since we're appending the locale slug, I double checked there isn't some _other_ URL param we're forgetting.

The standard step navigation code is here: https://github.com/Automattic/wp-calypso/blob/d8e03024b72b06190080c88a8c7c0f990eb6bb6c/client/landing/stepper/declarative-flow/internals/hooks/use-flow-navigation/index.tsx#L58

Looks like language is the only special URL fragment. There's query params to consider, but IMO dropping any query makes sense. We're navigating to a brand new flow, I think it makes sense to drop query params.

**Pull request merged**